app/scripts/migrations/084.ts

```diff
@@ -25,9 +25,26 @@ export async function migrate(originalVersionedData: {
 function transformState(state: Record<string, unknown>) {
   if (
     !hasProperty(state, 'NetworkController') ||
-    !isObject(state.NetworkController) ||
-    !hasProperty(state.NetworkController, 'network')
+    !isObject(state.NetworkController)
   ) {
+    global.sentry?.captureException?.(
+      new Error(
+        `typeof state.NetworkController is ${typeof state.NetworkController}`,
+      ),
+    );
+    return state;
+  }
+  if (!hasProperty(state.NetworkController, 'network')) {
+    const inPost077SupplementFor084state =
+      state.NetworkController.networkId !== undefined;
+    if (!inPost077SupplementFor084state) {
```

Comment:
Nit: This was a little confusing due to the double negative. Could we flip this? i.e. check that it's undefined, rather than that it's "not not undefined"

Reply:
addressed in 1d4e534159